ts/features/fci/saga/networking/handleCreateFilledDocument.ts

```diff
@@ -37,3 +54,47 @@ export function* handleCreateFilledDocument(
     yield* put(fciLoadQtspFilledDocument.failure(getNetworkError(e)));
   }
 }
+
+/**
+ * Handle the FCI polling saga
+ * This saga is used to poll the QTSP filled_document to check if the
+ * document is ready to be downloaded by the user. The polling is done
+ * every POLLING_FREQ_TIMEOUT seconds and the polling is stopped when
+ * the document is ready or when the polling is stopped if
+ * the polling time exceeds POLLING_TIME_THRESHOLD.
+ */
+export function* watchFciPollSaga(): SagaIterator {
+  const qtspFilledDocumentUrl = yield* select(fciQtspFilledDocumentUrlSelector);
+  if (qtspFilledDocumentUrl) {
+    const startPollingTime = new Date().getTime();
+    while (true) {
+      try {
+        const response = yield* call(fetch, qtspFilledDocumentUrl);
+        const responseStatus = response.status;
+        if (responseStatus === 200) {
+          yield* put(
+            fciPollFilledDocument.success({
+              isReady: true
+            })
+          );
+          yield* put(fciCancelPollingFilledDocument());
+        }
+        yield* call(startTimer, POLLING_FREQ_TIMEOUT);
+        const now = new Date().getTime();
+        if (now - startPollingTime >= POLLING_TIME_THRESHOLD) {
+          throw Error("Polling time exceeded");
+        }
```

Comment:
@francescopersico I have removed the retryTimes from the store to follow your suggestion. I used a time threshold to track the max retries the saga should do. It's the same approach used here https://github.com/pagopa/io-app/blob/f4957e769b25799390e6478d40730a53fa9d9702/ts/features/bonus/cgn/saga/networking/activation/getBonusActivationSaga.ts#L134

Reply:
@francescopersico What do you think about this changes?